aiida/orm/entities.py

```diff
@@ -12,21 +12,40 @@
 from __future__ import print_function
 from __future__ import absolute_import
 
+import typing
+
 from aiida.common import exceptions
+from aiida.common import datastructures
 from aiida.common.utils import classproperty, type_check
+from aiida.manage import get_manager
 
 __all__ = ('Entity', 'Collection')
 
+EntityType = typing.TypeVar('EntityType')  # pylint: disable=invalid-name
+
 
-class Collection(object):
+class Collection(typing.Generic[EntityType]):
     """Container class that represents the collection of objects of a particular type."""
 
+    # A store for any backend specific collections that already exist
+    _COLLECTIONS = datastructures.LazyStore()
+
+    @classmethod
+    def get_collection(cls, entity_type, backend):
+        """
+        Get the collection for a given entity type and backend instance
+
+        :param entity_type: the entity type e.g. User, Computer, etc
+        :param backend: the backend instance to get the collection for
+        :return: the collection instance
+        """
+        # Lazily get the collection i.e. create only if we haven't done so yet
+        return cls._COLLECTIONS.get((entity_type, backend), lambda: entity_type.Collection(backend, entity_type))
+
     def __init__(self, backend, entity_class):
+        """Construct a new entity collection"""
         # assert issubclass(entity_class, Entity), "Must provide an entity type"
```

Comment:
Should we uncomment this assert, or if not just remove it

Reply:
Uncommented